stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/BaseMessageListHeaderViewModel.kt

```diff
@@ -1,70 +1,99 @@
 package com.getstream.sdk.chat.viewmodel
 
 import androidx.lifecycle.LiveData
-import androidx.lifecycle.MediatorLiveData
 import androidx.lifecycle.MutableLiveData
-import androidx.lifecycle.Transformations.map
 import androidx.lifecycle.ViewModel
-import io.getstream.chat.android.client.logger.ChatLogger
+import androidx.lifecycle.asLiveData
+import androidx.lifecycle.viewModelScope
+import com.getstream.sdk.chat.viewmodel.messages.MessageListViewModel
+import io.getstream.chat.android.client.ChatClient
 import io.getstream.chat.android.client.models.Channel
 import io.getstream.chat.android.client.models.Member
 import io.getstream.chat.android.client.models.Message
 import io.getstream.chat.android.client.models.User
 import io.getstream.chat.android.core.internal.InternalStreamChatApi
-import io.getstream.chat.android.livedata.ChatDomain
+import io.getstream.chat.android.offline.experimental.channel.state.ChannelState
+import io.getstream.chat.android.offline.experimental.extensions.asReferenced
+import io.getstream.chat.android.offline.experimental.extensions.globalState
+import io.getstream.chat.android.offline.experimental.global.GlobalState
 import io.getstream.chat.android.offline.model.ConnectionState
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.map
 
+/**
+ * The ViewModel used by [MessageListHeaderView].
+ *
+ * @param cid The CID of the current channel.
+ * @param chatClient An instance of the low level chat client.
+ * @param globalState Global state of OfflinePlugin. Contains information
+ * such as the current user, connection state, unread counts etc.
+ */
 public abstract class BaseMessageListHeaderViewModel @InternalStreamChatApi constructor(
     cid: String,
-    private val chatDomain: ChatDomain,
+    chatClient: ChatClient = ChatClient.instance(),
+    globalState: GlobalState = chatClient.globalState,
 ) : ViewModel() {
 
-    private val _activeThread = MutableLiveData<Message?>()
-    private val _members = MediatorLiveData<List<Member>>()
-    private val _channelState = MediatorLiveData<Channel>()
-    private val _anyOtherUsersOnline = MediatorLiveData<Boolean>()
-    private val _typingUsers = MediatorLiveData<List<User>>()
+    /**
+     * Holds information about the current channel and is actively updated.
+     */
+    public val channelState: ChannelState =
+        chatClient.asReferenced().watchChannel(cid, MessageListViewModel.DEFAULT_MESSAGES_LIMIT).asState(viewModelScope)
```

Comment:
Do we even need a concrete message limit when we watch channels in `ViewModels` that deal with headers, etc. and do not display the actual messages?

Reply:
We don't need any messages here. We should replace it with `0`